src/Workspaces/VisualBasic/Portable/CodeCleanup/Providers/ReduceTokensCodeCleanupProvider.vb

```diff
@@ -222,26 +264,75 @@ Namespace Microsoft.CodeAnalysis.CodeCleanup.Providers
+                        End If
+                        Return $"&O{ConvertToOctalString(val1)}"
+                    Case Else
+                        Throw ExceptionUtilities.Unreachable
+                End Select
+            End Function
+
+            Private Shared Function CreateLiteralToken(token As SyntaxToken, newValueString As String, newValue As Object) As SyntaxToken
                 ' create a new token with valid token text and carries over annotations attached to original token to be a good citizen 
                 ' it might be replacing a token that has annotation injected by other code cleanups
                 Dim leading = If(token.LeadingTrivia.Count > 0, token.LeadingTrivia, SyntaxTriviaList.Create(SyntaxFactory.ElasticMarker))
                 Dim trailing = If(token.TrailingTrivia.Count > 0, token.TrailingTrivia, SyntaxTriviaList.Create(SyntaxFactory.ElasticMarker))
 
-                Return token.CopyAnnotationsTo(SyntaxFactory.FloatingLiteralToken(leading, newValueString, token.GetTypeCharacter(), newValue, trailing))
+                Select Case token.Kind
+                    Case SyntaxKind.FloatingLiteralToken
+                        Return token.CopyAnnotationsTo(SyntaxFactory.FloatingLiteralToken(leading, newValueString, token.GetTypeCharacter(), DirectCast(newValue, Double), trailing))
+                    Case SyntaxKind.DecimalLiteralToken
+                        Return token.CopyAnnotationsTo(SyntaxFactory.DecimalLiteralToken(leading, newValueString, token.GetTypeCharacter(), DirectCast(newValue, Decimal), trailing))
+                    Case SyntaxKind.IntegerLiteralToken
+                        Return token.CopyAnnotationsTo(SyntaxFactory.IntegerLiteralToken(leading, newValueString, token.GetBase().Value, token.GetTypeCharacter(), DirectCast(newValue, ULong), trailing))
+                    Case Else
+                        Throw ExceptionUtilities.Unreachable
+                End Select
             End Function
 
-            Private Shared Function CreateDecimalLiteralToken(token As SyntaxToken, newValueString As String, newValue As Decimal) As SyntaxToken
-                Debug.Assert(token.Kind = SyntaxKind.DecimalLiteralToken)
+            Private Function ConvertToOctalString(value As ULong) As String
+                Dim exponent As ULong = value
+                Dim builder As New StringBuilder()
 
-                ' create a new token with valid token text and carries over annotations attached to original token to be a good citizen 
-                ' it might be replacing a token that has annotation injected by other code cleanups
-                Dim leading = If(token.LeadingTrivia.Count > 0, token.LeadingTrivia, SyntaxTriviaList.Create(SyntaxFactory.ElasticMarker))
-                Dim trailing = If(token.TrailingTrivia.Count > 0, token.TrailingTrivia, SyntaxTriviaList.Create(SyntaxFactory.ElasticMarker))
+                While (exponent > 0)
```

Comment:
@jasonmalinowski yes that will fail at the moment. I'll fix it and add it into the tests to cover that case